daemon/redis.c

```diff
@@ -62,9 +61,23 @@ static int redisCommandNR(redisContext *r, const char *fmt, ...)
 #define STR(x) (x)->s, (size_t) (x)->len
 #define STR_R(x) (x)->str, (size_t) (x)->len
 #define S_LEN(s,l) (s), (size_t) (l)
+#define STRSTR(x) (x)->s
 
 #endif
 
+static void redis_restore_call(struct redis *r, struct callmaster *m, const redisReply *id, enum call_type);
+static int redis_check_conn(struct redis *r);
+static void json_restore_call(struct redis *r, struct callmaster *m, redisReply *id, enum call_type type);
+
+static int str_cut(char *str, int begin, int len) {
```

Comment:
Looking at the instances where this function is used, I believe it should be removed. It needlessly modifies the string in place. The appropriate replacement would be to use `str_shift` or something equivalent for `redisReply` objects. I think the best way to go would be to have a `str_init_redis` function to initialize a `str` object from a `redisReply` object and the manipulate the `str` object (for example with `str_shift`). There already are a few instances of `str_init_len` in the code, used to do exactly that. They should also be replaced with a call to something like this:

```
INLINE void str_init_redis(str *out, const redisReply *rr) {
  str_init_len(out, rr->str, rr->len);
}
```